plugins/module_utils/vm.py

```diff
@@ -1099,7 +1103,33 @@ def _delete_not_used_disks(cls, module, rest_client, vm, changed, disk_key):
                     ),
                     module.check_mode,
                 )
-                TaskTag.wait_task(rest_client, task_tag, module.check_mode)
+                try:
+                    TaskTag.wait_task(rest_client, task_tag, module.check_mode)
+                except errors.TaskTagError as ex:
+                    # Delete failed, maybe because VM was running and disk was in use.
+                    # If VM is running, shutdown VM and retry delete.
+                    if ex.task_tag_state != "ERROR":
+                        raise
+                    # The particular "formattedMessage" is returned by HyperCore 9.2.17
+                    if (
+                        ex.task_tag["formattedMessage"]
+                        != "Unable to delete block device from VM '%@': Still in use"
+                    ):
+                        raise
+                    vm_fresh_data = rest_client.get_record(
+                        f"/rest/v1/VirDomain/{vm.uuid}", must_exist=True
+                    )
+                    if vm_fresh_data["state"] != "RUNNING":
+                        raise
+                    # shutdown and retry remove
+                    vm.do_shutdown_steps(module, rest_client)
```

Comment:
I'm not clear on the full context for this call. Is the user aware that the VM may be shutdown by attempting to delete a disk? Would it be better to leave this in the user's hands? In other words, if the VM needs to be shutdown before deleting the disk, should the user be required to include that in their playbooks? 

I see the previously existing call to shutdown on line 1099. That implies to me that the shutdown is OK here but I wanted to ask the question. Is the shutdown call only honored if `force_reboot: True` or something?

Reply:
User is aware module will try to shutdown VM if this is required to remove a disk: Docs:
```
  - Module tries to remove disks from a running VM.
    If disk cannot be removed from running VM,
    then VM will be shutdown, disk will be removed, and VM is started back.
  - VM has C(shutdown_timeout) time to respond to shutdown request.
    If VM is not shutoff within I(shutdown_timeout),
    then a force shutdown will be issued if C(force_reboot=True).
```

The `force_reboot` could be called `allow_force_reboot_if_nice_shutdown_does_not_work`.